js_modules/dagit/packages/core/src/ui/SectionedLeftNav.tsx

```diff
@@ -112,15 +120,61 @@ interface SectionProps {
   expanded: boolean;
   onToggle: (repoAddress: RepoAddress) => void;
   option: DagsterRepoOption;
+  pathJobMatch: string | null;
+  repoAddress: RepoAddress;
   showRepoLocation: boolean;
 }
 
 export const Section: React.FC<SectionProps> = (props) => {
-  const {expanded, onToggle, option, showRepoLocation} = props;
-  const repoAddress = buildRepoAddress(option.repository.name, option.repositoryLocation.name);
+  const {expanded, onToggle, option, pathJobMatch, repoAddress, showRepoLocation} = props;
   const jobItems = React.useMemo(() => getJobItemsForOption(option), [option]);
+
   const anyJobs = jobItems.length > 0;
   const showJobs = expanded && anyJobs;
+  const match = pathJobMatch && jobItems.find((job) => job.name === pathJobMatch);
+  const matchRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    if (match && matchRef.current) {
+      matchRef.current.scrollIntoView({block: 'nearest'});
+    }
+  }, [match]);
+
+  const visibleJobs = () => {
+    if (showJobs) {
+      return (
+        <Box
+          padding={{vertical: 8, horizontal: 12}}
+          border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+        >
+          {jobItems.map((jobItem) => {
+            const active = pathJobMatch === jobItem.name;
+            return (
+              <JobItem
+                job={jobItem}
+                key={jobItem.path}
+                ref={active ? matchRef : undefined}
+                active={active}
+              />
+            );
+          })}
+        </Box>
+      );
+    }
+
+    if (match) {
+      return (
+        <Box
+          padding={{vertical: 8, horizontal: 12}}
+          border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+        >
+          <JobItem job={match} active ref={matchRef} />
+        </Box>
+      );
+    }
+
```

Comment:
Hmm I get the need for this but it feels a bit unexepected!

Reply:
We don't have to, it was just suggested on a Slack thread when discussing this situation. Apparently this is what Slack does when viewing a channel that is in a collapsed section.